src/help/components/landing/HelpQuickSearch.tsx

```diff
@@ -0,0 +1,122 @@
+
+import { help as helpURL } from '../../../shared/config/apiUrls';
+import { LocationToPath, Location } from '../../../app/config/urls';
+
+import { HelpSearchResponse } from '../../adapters/helpConverter';
+
+import styles from './styles/help-quick-search.module.scss';
+
+const numberResultsInView = 5 as const;
+
+type LocationState = { query: string };
+
+const HelpQuickSearch = () => {
+  const location = useLocation<undefined | LocationState>();
+  const history = useHistory();
+
+  const query = location?.state?.query || '';
+  const [searchValue, setSearchValue] = useState<string>(query);
+  const dataObject = useDataApiWithStale<HelpSearchResponse>(
+    query && helpURL.search({ query })
+  );
+  const replaceQueryInHistory = useMemo(
+    () =>
+      debounce((searchValue: string) => {
+        history.replace({
+          pathname: LocationToPath[Location.HelpResults],
+          state: {
+            query: searchValue || undefined,
+          },
+        });
+      }, 500),
+    [history]
+  );
+
+  useEffect(() => {
+    replaceQueryInHistory(searchValue);
+    return replaceQueryInHistory.cancel;
+  }, [replaceQueryInHistory, searchValue]);
+
+  const allArticlesLocation = useMemo(
+    () => ({
+      pathname: LocationToPath[Location.HelpResults],
+      search: qs.stringify({ query }),
+    }),
+    [query]
+  );
+
+  const handleKeyDown = useCallback(
+    ({ key }: KeyboardEvent<HTMLInputElement>) => {
+      if (key === 'Enter') {
+        history.push(allArticlesLocation);
+      }
+    },
+    [allArticlesLocation, history]
+  );
+
+  const allArticles = dataObject?.data?.results;
+  const infoData = allArticles
+    ?.slice(0, numberResultsInView)
+    .map(({ matches, title, id }) => {
+      const titleMatch = matches?.title?.[0];
+      const contentMatch = matches?.content?.[0];
+      const to = generatePath(LocationToPath[Location.HelpEntry], {
+        accession: id,
+      });
+      return {
+        title: (
+          <Link to={to}>
+            {titleMatch ? <CleanHighlightMarkDown md={titleMatch} /> : title}
+          </Link>
+        ),
+        content: contentMatch && <CleanHighlightMarkDown md={contentMatch} />,
+        to,
+      };
+    });
+
+  return (
+    <div className={styles['help-autocomplete']}>
+      <SearchInput
+        isLoading={dataObject.loading}
+        onChange={(event: ChangeEvent<HTMLInputElement>) =>
+          setSearchValue(event.target.value)
+        }
+        placeholder="Search"
+        value={searchValue}
+        onKeyDown={handleKeyDown}
+        autoFocus
+      />
+      {!!allArticles?.length && !!infoData?.length && searchValue && (
+        <Card>
+          <InfoList
+            className={styles['help-autocomplete__result-list']}
+            infoData={infoData}
+          />
+          <div className={styles['help-autocomplete__all-link']}>
+            <Link to={allArticlesLocation}>
+              Show all results ({allArticles.length})
```

Comment:
Should we not show this if there's only 1 result?

Reply:
I think this refers to the last thread on the Slack #ux channel so options have been discussed there already

Reply:
I don't think there's any point in going "full view" for 1 result as the facets don't add any functionality in that case...

Reply:
My thinking was to do it for consistency as there's only two ways to get to the full results:

1. Pushing enter
2. Clicking on the full view link

Route 1 may not always be obvious and it could be frustrating if the user doesn't always see the link because of the number of results.

Reply:
There's only 1 result though, so in the user workflow nowhere else to go than to that entry?

Reply:
I'm thinking of the instance when the user is returned a single result which isn't what they're looking for and they're thinking "I want the full view with the categories on the side - how do I get there? There used to be a link at the bottom in my other searches."

Reply:
I wouldn't be too attached to one approach or the other. Once either is implemented, we can easily play with it, have a feel, find edge-case examples, and discuss in a design meeting. This is easily changeable one way or the other if needed

Reply:
👌